### PR TITLE
tests: set APPLICATION from folder name

### PR DESCRIPTION
**Pull request by kaspar030**
There where a some tests where APPLICATION did not match the folder name (plus "test_"):

```
[kaspar@booze tests (master)]$ for i in */Makefile; do test "$(dirname $i)" = "$(grep APPLICATION $i | cut -d ' ' -f 3)" || echo $i; done
conn_can/Makefile
driver_pcd8544/Makefile
events/Makefile
evtimer_underflow/Makefile
gnrc_netif/Makefile
mcuboot/Makefile
netstats_l2/Makefile
pkg_cmsis-dsp/Makefile
pthread_condition_variable/Makefile
slip/Makefile
xtimer_usleep_short/Makefile
[kaspar@booze tests (master)]$
```

Most where either c&p leftovers or typos. Some just had a different name, but using the folder name still makes sense.

This PR sets ```APPLICATION``` in ```tests/Makefile.tests_common``` to the test's folder name.
Another commit removes the now redundant name definitions from the actual tests' Makefiles.

### CODE REVIEW

File: tests/Makefile.tests_common [hunk 1]
```diff
@@ -1,7 +1,7 @@
-ifneq (,$(filter driver_%,$(APPLICATION)))
+APPLICATION ?= tests_$(shell basename $(CURDIR))
```
Comment (gebart) [inline tests/Makefile.tests_common]: I think you should be able to do `$(not directly $(CURDIR)) ` or some variant thereof to avoid calling basename

Comment (kaspar030) [reply tests/Makefile.tests_common]: Yes, nice! Changed.

(will have to also use that to where we set the default module name)

File: tests/Makefile.tests_common [hunk 2]
```diff
@@ -1,7 +1,7 @@
-ifneq (,$(filter driver_%,$(APPLICATION)))
+APPLICATION ?= tests_$(notdir $(patsubst %/,%,$(CURDIR)))
```
Comment (cgundogan) [inline tests/Makefile.tests_common]: Should we be stricter here and **not** allow another `APPLICATION` definition in the tests? IMO this ensures consistency .. `make` should fail if you remove the `?` and a test application defines its own `APPLICATION`, right?

Comment (kaspar030) [reply tests/Makefile.tests_common]: A good idea for the next step! ;)

There are a some tests (openthread, lwmac) which do not even use tests/Makefile.tests_common. We can unify them (as followup), then force as you suggested. But not in this PR.

Comment (cgundogan) [reply tests/Makefile.tests_common]: >A good idea for the next step! ;)

I'll take your word for that (:

**Pull request merged**